components/query/query-base.service.ts

```diff
@@ -0,0 +1,403 @@
+    'measure',
+    'composition_browser',
+    'analysis',
+    'sensors',
+    'draw',
+  ];
+  last_coordinate_clicked: any;
+  hoverPopup: any;
+  getFeatureInfoStarted: Subject<any> = new Subject();
+  getFeatureInfoCollected: Subject<any> = new Subject();
+  queryStatusChanges: Subject<boolean> = new Subject();
+  vectorSelectorCreated: Subject<Select> = new Subject();
+
+  constructor(
+    private HsMapService: HsMapService,
+    private HsConfig: HsConfig,
+    private HsLayoutService: HsLayoutService,
+    private HsUtilsService: HsUtilsService,
+    private HsEventBusService: HsEventBusService,
+    private DomSanitizer: DomSanitizer
+  ) {
+    this.vectorSelectorCreated.subscribe((selector) => {
+      this.selector = selector;
+    });
+
+    this.HsMapService.loaded().then(() => this.init());
+  }
+  /**
+   *
+   */
+  init() {
+    this.map = this.HsMapService.map;
+    this.activateQueries();
+    this.map.on('singleclick', (evt) => {
+      this.HsEventBusService.mapClicked.next(
+        Object.assign(evt, {
+          coordinates: this.getCoordinate(evt.coordinate),
+        })
+      );
+      if (!this.queryActive) {
+        return;
+      }
+      this.popupClassname = '';
+      if (!this.dataCleared) {
+        this.clearData();
+      }
+      this.dataCleared = false;
+      this.currentQuery = (Math.random() + 1).toString(36).substring(7);
+      this.setData(this.getCoordinate(evt.coordinate), 'coordinates', true);
+      this.last_coordinate_clicked = evt.coordinate; //It is used in some examples and apps
+      this.data.selectedProj = this.data.coordinates[0].projections[0];
+      this.getFeatureInfoStarted.next(evt);
+    });
+
+    if (this.HsConfig.popUpDisplay && this.HsConfig.popUpDisplay === 'hover') {
+      this.map.on(
+        'pointermove',
+        this.HsUtilsService.debounce(this.showPopUp, 500, false, this)
+      );
+    } else if (
+      this.HsConfig.popUpDisplay &&
+      this.HsConfig.popUpDisplay === 'click'
+    ) {
+      this.map.on(
+        'singleclick',
+        this.HsUtilsService.debounce(this.showPopUp, 500, false, this)
+      );
+    } /* else none */
+  }
+
+  /**
+   * @param e Event, which triggered this function
+   */
+  showPopUp(e) {
+    if (e.dragging) {
+      return;
+    }
+    const map = e.map;
+    this.featuresUnderMouse = map.getFeaturesAtPixel(e.pixel);
+    if (this.featuresUnderMouse !== null) {
+      this.featuresUnderMouse = this.featuresUnderMouse.filter((feature) => {
+        return (
+          feature.getLayer &&
+          feature.getLayer(map) &&
+          feature.getLayer(map).get('title').length > 0 &&
+          feature.getLayer(map).get('title') !== 'Point clicked'
+        );
+      });
+      this.featureLayersUnderMouse = this.featuresUnderMouse.map((f) =>
+        f.getLayer(this.HsMapService.map)
+      );
+      this.featureLayersUnderMouse = this.HsUtilsService.removeDuplicates(
+        this.featureLayersUnderMouse,
```

Comment:
Pop up seems to be working.... we just might also filter out features that doesnt have `.get('popUp') `...  (thats where popUp content come from, right?) so we doesnt show empty popup on drawn features or features that come from layer not configured to show pop-ups

Reply:
Well, `.get('popUp')` might be used, but on a `Layer` level. The attributes to be displayed are described there, not on a feature level.

Reply:
Sure, freatures where feature.getLayer(map).get('popUp') is false

Reply:
I resolved it here: https://github.com/hslayers/hslayers-ng/pull/965/commits/e5909c3439e7343e0fcd51d58fd3d74313f8b0d3